modules/containers/include/hephaestus/containers/bit_flag.h

```diff
@@ -66,6 +82,22 @@ class BitFlag {
                   "Enum is not valid for BitFlag, its values must be power of 2.");
   }
 
+  /// Constructs a BitFlag with the given underlying value.
+  ///
+  /// An 'InvalidParameterException' is thrown if there are bits set in 'underlying_value' which do not
+  /// correspond to a valid enum value.
+  constexpr explicit BitFlag(std::underlying_type_t<EnumT> underlying_value) : value_(underlying_value) {
```

Comment:
nit. we define `T` as the underlying type so we could use it here. If you think the name is not expressive enough, feel free to refactor to a better one